controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1433,6 +1453,20 @@ func (r *BareMetalHostReconciler) doServiceIfNeeded(prov provisioner.Provisioner
 		dirty = true
 	}
 
+	if hfcDirty && started {
+		hfcDirty, err = r.saveHostFirmwareComponents(prov, info, hfc)
+		if err != nil {
+			return actionError{errors.Wrap(err, "could not save the host firmware components")}
+		}
+
+		if hfcDirty {
+			if err := r.Status().Update(info.ctx, hfc); err != nil {
```

Comment:
Please remind me, why are we doing this? It looks like the status changes are actually in the HFC object, not in the BMH. And the BMH will be saved by the higher level controller code anyway.

Reply:
When calling the saveHostFirmwareComponents we change the information there, so the call here is to indeed update the CR

Reply:
Right, I somehow thought you were updating the BMH